From 6c72dc3c40bfb5ec5bd8511f4fc1f31540c433e2 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Fri, 26 Jan 2024 18:10:53 +0100
Subject: [PATCH 0586/1695] dts: stm32: Update over-current protection on
 stm32mp257f-dk

Update over-current protection on stm32mp257f-ev1:
Set BUCK3, BUCK7, LDO2, LDO4, LDO7, LDO8 in HICUP mode to
handle OCP interruption with an error message.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 0b247d48f3fa0d61f86430d0d8a351d689ccaa3d dts: stm32: Update over-current protection on stm32mp257f-dk]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: Iec804a6afa8a45efb337eaab797242d00a62a586
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/354905
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Tested-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/arch/arm/dts/stm32mp257f-dk.dts | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp257f-dk.dts b/core/arch/arm/dts/stm32mp257f-dk.dts
index c93cb2073..734377d5a 100644
--- a/core/arch/arm/dts/stm32mp257f-dk.dts
+++ b/core/arch/arm/dts/stm32mp257f-dk.dts
@@ -116,6 +116,7 @@
 				regulator-name = "vddgpu_pmic";
 				regulator-min-microvolt = <800000>;
 				regulator-max-microvolt = <900000>;
+				regulator-over-current-protection;
 			};
 			vddio_pmic: buck4 {
 				regulator-name = "vddio_pmic";
@@ -196,6 +197,7 @@
 				regulator-min-microvolt = <3300000>;
 				regulator-max-microvolt = <3300000>;
 				regulator-always-on;
+				regulator-over-current-protection;
 				st,pwrctrl_sel = <1>;
 				st,pwrctrl_enable;
 
@@ -225,6 +227,7 @@
 				regulator-name = "vdd_emmc";
 				regulator-min-microvolt = <3300000>;
 				regulator-max-microvolt = <3300000>;
+				regulator-over-current-protection;
 				st,pwrctrl_sel = <3>;
 				st,pwrctrl_reset;
 			};
@@ -257,6 +260,7 @@
 				regulator-min-microvolt = <3300000>;
 				regulator-max-microvolt = <3300000>;
 				regulator-always-on;
+				regulator-over-current-protection;
 			};
 			v5v_hdmi: ldo5 {
 				regulator-name = "v5v_hdmi";
@@ -279,6 +283,7 @@
 				regulator-name = "vdd_sdcard";
 				regulator-min-microvolt = <3300000>;
 				regulator-max-microvolt = <3300000>;
+				regulator-over-current-protection;
 				st,regulator-bypass-microvolt = <3300000>;
 				st,pwrctrl_sel = <3>;
 				st,pwrctrl_reset;
@@ -287,6 +292,7 @@
 				regulator-name = "vddio_sdcard";
 				regulator-min-microvolt = <1800000>;
 				regulator-max-microvolt = <3300000>;
+				regulator-over-current-protection;
 				st,regulator-bypass-microvolt = <3300000>;
 				st,pwrctrl_sel = <3>;
 				st,pwrctrl_reset;
-- 
2.47.2

